src/components/Select.jsx

```diff
@@ -222,7 +207,8 @@ export default class Select extends Component {
     let focusedOptionValue;
 
     if (properties.children) {
-      this.children = flatten(properties.children);
+      this.children = flattenReactChildren(properties.children);
+      this.options = filter(this.children, isOption);
     }
```

Comment:
I added options to 'this' coz I found that filtering was done at a lot of places, this can save us from that.